### PR TITLE
freeciv: 2.6.6 -> 3.0.1

### PR DESCRIPTION
**Pull request by cmm**
###### Things done

- Built on platform(s)
  - [v] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [v] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by QuantMint** [comment]
Result of `nixpkgs-review pr 177222` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>freeciv</li>
    <li>freeciv_gtk</li>
    <li>freeciv_qt</li>
  </ul>
</details>

File: pkgs/games/freeciv/default.nix [hunk 1]
```diff
@@ -1,6 +1,6 @@
 { lib, stdenv, fetchFromGitHub, autoreconfHook, lua5_3, pkg-config, python3
-, zlib, bzip2, curl, xz, gettext, libiconv
-, sdlClient ? true, SDL, SDL_mixer, SDL_image, SDL_ttf, SDL_gfx, freetype, fluidsynth
+, zlib, bzip2, curl, xz, gettext, libiconv, icu
+, sdlClient ? true, SDL2, SDL2_mixer, SDL2_image, SDL2_ttf, SDL2_gfx, freetype, fluidsynth
```
Comment (QuantMint) [inline pkgs/games/freeciv/default.nix:3]: The sdl2 client seem to be marked as experimental on the website, in fact it's quite buggy on sway. Maybe it's better to enable the gtk3 client instead or both of them simultaneously?

Comment (cmm) [reply pkgs/games/freeciv/default.nix:3]: none of the clients are enabled simultaneously, I think?
"freeciv" is sdl2 (formerly sdl), "freeciv_gtk" is gtk3, "freeciv_qt" is qt5, all different derivations, just as it was before.

Did the sdl client work well before they moved to sdl2?  I wouldn't know, I have no idea who uses that or why it even exists.

Comment (QuantMint) [reply pkgs/games/freeciv/default.nix:3]: > none of the clients are enabled simultaneously, I think? "freeciv" is sdl2 (formerly sdl), "freeciv_gtk" is gtk3, "freeciv_qt" is qt5, all different derivations, just as it was before.
> 
> Did the sdl client work well before they moved to sdl2? I wouldn't know, I have no idea who uses that or why it even exists.

I just think that the gtk3 client should be enabled by default

Comment (cmm) [reply pkgs/games/freeciv/default.nix:3]: I agree (that's the one I use), but:

- the gtk3 client is also pretty buggy
- I don't think changing defaults is in scope of a lowly upgrade PR

Comment (QuantMint) [reply pkgs/games/freeciv/default.nix:3]: > I agree (that's the one I use), but:
> 
>     * the gtk3 client is also pretty buggy
> 
>     * I don't think changing defaults is in scope of a lowly upgrade PR

This pr is already changing the default, from the sdl one to the sdl2 one(that is experimental).

Comment (cmm) [reply pkgs/games/freeciv/default.nix]: it's the upstream who pulled that rug, but you have a point.

**Comment by QuantMint** [comment]
I am currently getting this:
```
2: No real audio plugin present.
2: Proceeding with sound support disabled.
2: For sound support, install SDL2_mixer
2: http://www.libsdl.org/projects/SDL_mixer/index.html
```

**Comment by cmm** [comment]
huh, it does appear that all clients need sdl for sound, will fix.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -31803,18 +31803,25 @@ with pkgs;
   freeciv = callPackage ../games/freeciv {
     autoreconfHook = buildPackages.autoreconfHook269;
     qt5 = qt514;
```
[the remark this answers is not in the document]
Comment (cmm) [reply pkgs/top-level/all-packages.nix:31805]: does not appear to and the commit that introduced it did not explain why, I'll remove it

**Comment by QuantMint** [comment]
Result of `nixpkgs-review pr 177222` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>freeciv (freeciv_gtk)</li>
    <li>freeciv_qt</li>
    <li>freeciv_sdl2</li>
  </ul>
</details>

**Review by Tungsten842 — APPROVED** [review]

**Review by teto — APPROVED** [review]

**Pull request merged**